static/app/views/dashboardsV2/view.tsx

```diff
@@ -37,9 +39,18 @@ function ViewEditDashboard(props: Props) {
 
     const constructedWidget = constructWidgetFromQuery(location.query);
     setNewWidget(constructedWidget);
-    // Clean up url after constructing widget from query string
+    // Clean up url after constructing widget from query string, only allow GHS params
     if (constructedWidget) {
-      browserHistory.replace(location.pathname);
+      browserHistory.replace({
+        pathname: location.pathname,
+        query: {
+          ...Object.keys(location.query)
```

Comment:
I think we could use a lodash util here to pass through the `ALLOWED_PARAMS` if you wanted to be a bit more concise!
https://lodash.com/docs/4.17.15#pick
```js
query: pick(location.query, ALLOWED_PARAMS)
```

Reply:
nice, definitely much more concise!